File: src/frontend/src/lib/types/show.ts
import type { Season } from "./season";

export type ShowRequest = {
    ServiceTag: string;
    ShowID: string;
};

export type Show = {
    adult: boolean;
    backdropURL: string;
    id: string;
    seasonCount: number;
    name: string;
    originalName: string;
    overview: string;
    posterURL: string;
    mediaType: string;
    originalLanguage: string;
    genres: ShowGenre[];
    cast: string[];
    directors: string[];
    firstAirDate: string;
    originCountry: string;
    availabilityStatus: string;
    seasons: Season[];
};

export type ShowGenre = {
    id: string;
    name: string;
};

File: src/backend/internal/http/model/show.go
package model

import (
	"time"
)

// URL params
type ShowRequest struct {
	ServiceTag string
	ShowID     string
}

// Show Response
type Show struct {
	Adult              bool        `json:"adult"`
	BackdropURL        string      `json:"backdropURL"`
	ID                 string      `json:"id"`
	SeasonCount        int         `json:"seasonCount"`
	Name               string      `json:"name"`
	OriginalName       string      `json:"originalName"`
	Overview           string      `json:"overview"`
	PosterURL          string      `json:"posterURL"`
	MediaType          string      `json:"mediaType"`
	OriginalLanguage   string      `json:"originalLanguage"`
	Genres             []ShowGenre `json:"genres"`
	Cast               []string    `json:"cast"`
	Directors          []string    `json:"directors"`
	FirstAirDate       time.Time   `json:"firstAirDate"`
	OriginCountry      string      `json:"originCountry"`
	AvailabilityStatus string      `json:"availabilityStatus"`
	Seasons            []Season    `json:"seasons"`
}

type ShowGenre struct {
	ID   string `json:"id"`
	Name string `json:"name"`
}
